code/game/machinery/recharger.dm

```diff
@@ -179,6 +145,60 @@
 		return
 	icon_state = icon_state_idle
 
+/obj/machinery/recharger/proc/get_cell_from(obj/item/I)
+	if(!I)
+		return null
```

Comment:
Do you think the if check is necessary here? Because in `try_recharging_if_possible()` you already check if `charging` exists before the call to this proc, which then checks if `I`, aka `charging`, exists a second time.

Reply:
Well, I'd consider that an utility function in the context of the recharger and I personally think utils should be self-contained in terms of checks rather than assuming their input is sane, so I'd like to keep that in